### PR TITLE
Lovelace: Allow glance card to assume theme colors

### PR DESCRIPTION
**Pull request by thomasloven**
**Edit**: Documentation added: home-assistant/home-assistant.io#6506

This change lets the glance card act like [color-glance-card](https://github.com/thomasloven/lovelace-color-glance-card).

For now a sample config looks like this:
```yaml
- type: glance
  themed: true
  entities:
    - light.bed_light
```

I'm not sure about the name of the config variable, though, and as @balloob pointed out in discord, making it a boolean means values can't be added later.

Posting this as a WIP, hoping for ideas and advice.

### CODE REVIEW

**Comment by balloob** [comment]
I am not a fan of boolean config options as they don't usually hold well. Can we do something like: `theming: default/primary/secondary?`

**Comment by thomasloven** [comment]
`theming:` is a good idea, and extensible:
glance theming

**Comment by thomasloven** [comment]
Fewer choices, yet more...
theming2

```
- type: glance
  theming: primary
  entities: ...

- type: glance
  theming:
    background: rgba(0, 110, 128)
    text: white
    icons: var(--accent-color)
  entities: ...
```

**Comment by thomasloven** [comment]
just primary

```yaml
- type: glance
  title: default
  theming: default
  entities: ...

- type: glance
  title: primary
  theming: primary
  entities: ...

- type: glance
  title: none
  entities: ...
```

File: src/panels/lovelace/cards/hui-glance-card.js [hunk 1]
```diff
@@ -22,6 +22,12 @@ class HuiGlanceCard extends LocalizeMixin(EventsMixin(PolymerElement)) {
   static get template() {
     return html`
       <style>
+        :host(.theme-primary) {
+          --paper-card-background-color:var(--primary-color);
+          --paper-item-icon-color:var(--text-primary-color);
+          color:var(--text-primary-color);
+        }
+        :host(.theme-default) {}
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-glance-card.js]: No reason to add an empty theme

File: src/panels/lovelace/cards/hui-glance-card.js [hunk 2]
```diff
@@ -90,6 +96,11 @@ class HuiGlanceCard extends LocalizeMixin(EventsMixin(PolymerElement)) {
   setConfig(config) {
     this._config = config;
     this.updateStyles({ '--glance-column-width': (config && config.column_width) || '20%' });
+
+    if (config && config.theming && typeof(config.theming) === 'string') {
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-glance-card.js]: If it's not a string, we should throw an exception.

File: src/panels/lovelace/cards/hui-glance-card.js [hunk 3]
```diff
@@ -90,6 +96,11 @@ class HuiGlanceCard extends LocalizeMixin(EventsMixin(PolymerElement)) {
...
+        this.classList.add(`theme-${config.theming || 'default'}`);
```
Comment (balloob) [inline src/panels/lovelace/cards/hui-glance-card.js]: `|| default` is never used as you already checking for truthiness with `if (config.theming)`

**Comment by balloob** [comment]
Can you add the theming = primary to one of the demo glances? https://github.com/home-assistant/home-assistant-polymer/blob/master/gallery/src/demos/demo-hui-glance-card.js

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
:dolphin: :tada:

**Comment by iantrich** [comment]
image

**Pull request merged**